custom_metrics/privacy.js

```diff
@@ -1,24 +1,141 @@
+        } else {
+          tcData = {"error": "Failed to retrieve TCData"};
+        }
+      });
+    } else {
+      tcData = null;
+    }
+    return tcData;
+  })(),
+  iab_usp: (() => {
+    // IAB US Privacy User Signal Mechanism “USP API” is integrated on the page.
+    // https://github.com/InteractiveAdvertisingBureau/USPrivacy
+    // Test site: nfl.com
+    let uspData;
+    if (typeof __uspapi == 'function') {
+      // based on https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#gettcdata
+      __uspapi('getUSPData', 1, (result, success) => {
+        if(success) {
+          uspData = result;
+        } else {
+          uspData = {"error": "Failed to retrieve USPData"};
+        }
+      });
+    } else {
+      uspData = null;
+    }
+    return uspData;
+  })(),
+
+  // Ads Transparency Spotlight Data Disclosure schema
+  // Check `meta` tag cf. https://github.com/Ads-Transparency-Spotlight/documentation/blob/main/implement.md
+  // Only for top frame; can't access child frames (same-origin policy)
+  // Test site: ?
+  ads_transparency_spotlight: (() => {
+    let ats_meta_tag = document.querySelector('meta[name="AdsMetadata"]');
+    if (ats_meta_tag) {
+      return ats_meta_tag.content; // JSON message with metadata
+    } else {
+      return null;
+    }
+  })(),
+
+  // FLoC origin trial
+  // Check `meta` tag cf. https://developer.chrome.com/blog/floc/ (complement with relevant header later on)
+  // Test site: https://floc-ot-meta.glitch.me/
+  floc_origin_trial: (() => {
+    let floc_meta_tag = document.querySelector('meta[http-equiv="origin-trial"]');
+    if (floc_meta_tag) {
+      return floc_meta_tag.content; // Get origin trial token, could be useful to check if multiple enrolled sites are from the same webmaster
+    } else {
+      return null;
+    }
+  })(),
+
+  // Check function/variable accesses through string searches (wrappers cannot be used, as the metrics are only collected at the end of the test)
+  // FLoc - test site: https://floc.glitch.me/
+  floc_document_interestCohort: findStringInBodies('document.interestCohort'),
+  floc_interestCohort: findStringInBodies('interestCohort'),
+  // DNT - test site: https://www.theverge.com/
+  navigator_doNotTrack: findStringInBodies('navigator.doNotTrack'),
+  doNotTrack: findStringInBodies('doNotTrack'),
+  // GPC - test site: https://global-privacy-control.glitch.me/
+  navigator_globalPrivacyControl: findStringInBodies('navigator.globalPrivacyControl'),
+  globalPrivacyControl: findStringInBodies('globalPrivacyControl'),
+  // sensitive resources - accesses permissions policy
+  document_permissionsPolicy: findStringInBodies('document.permissionsPolicy'),
+  permissionsPolicy: findStringInBodies('permissionsPolicy'),
+  // user media (camera, microphone)
+  navigator_mediaDevices_getUserMedia: findStringInBodies('navigator.mediaDevices.getUserMedia'),
```

Comment:
@VictorLeP 
```suggestion
  navigator_mediaDevices_getUserMedia: findStringInBodies('navigator.mediaDevices.enumerateDevices'),
```
'getUserMedia' prompts the user to give permissions to access the stream.
I would rather use `enumerateDevices` which silently reads device info.

Reply:
I would do both then: `enumerateDevices` sounds more useful for fingerprinting, while `getUserMedia` provides access to the sensitive resource (i.e., being able to capture your webcam feed).

Reply:
I might include `getDisplayMedia` as well then.

Reply:
Updated in a17248f